### ISSUE TITLE
Set Pin Does not work

### ISSUE THREAD

**Issue by masterial** [getZION/app#82]
Set Pin functionality is broken in the app.

**Comment by Entryist** [comment]
I was able to set a PIN fine on a newly reset node. Please provide instructions to reproduce the issue.

**Comment by Entryist** [comment]
For clarity: Clicking "Set Pin" on the Security screen does nothing. Setting PIN during onboarding works fine.

**Comment by mohamedsamara** [comment]
@ajay-kays FYI set pin requires the user to re-export their backup keys. The pin is used in generating the[ backup key](https://github.com/getZION/app/blob/main/src/views/account/Security.tsx#L79).I  recommend take a look at how the pin is set for a new user, I think it's using the storage. [PIN component](https://github.com/getZION/app/blob/main/src/views/utils/pin.tsx)

**Comment by ajay-kays** [comment]
You mean whenever the pin is updated we need to make user export the key or we can just show a message saying "please update your new keys since pin is changed" ?

**Comment by mohamedsamara** [comment]
@ajay-kays no, it needs to be mandatory. User have to re-export the backup keys. I think we have the screen where they can't exit out the pin in the onboarding components. Also, the user have to enter the old pin before they make a new one. 

So this will be:
1. Set pin pressed.
2. Enter old pin. [How Pin is used](https://github.com/getZION/app/blob/main/src/views/onboard/choosePIN.tsx)
3. Enter new pin.
4. Re-export the keys. This will be: Enter new pin and backup key is copied to clipboard. Like [here](https://github.com/getZION/app/blob/main/src/views/account/Security.tsx#L79).

Does that make sense? Let me know if you have any questions, I'm happy to help!

**Comment by ajay-kays** [comment]
While login i don't see that user is being mandate to export the key. I had to go to profile section >> security >> backup keys to backup.
How do we mandate them cause i don't find a flow.

What i can do after pin changed i can remove the back button and disable gesture based back and show a message mentioning please export the keys.

Does that sounds good?

**Comment by Entryist** [comment]
Thanks Ajay, yes if you could knock out this issue too - https://github.com/getZION/app/issues/77 - we want to force the user to backup their key in the onboarding process.

**Comment by ajay-kays** [comment]
Thanks. I'm on it.
@mohamedsamara  Do have a component in which we can show notice message or some warning for highlighting purpose

**Comment by ajay-kays** [comment]
@Entryist  For now i'm using the Alert component for showing the message saying backup your new key. I'll be raising a PR in some time as I'm almost done with all changes for issue #82 & #74 with mandating user to backup the key while changing or setting new the pin

**Comment by mohamedsamara** [comment]
@ajay-kays for the above questions:

The solution we need is to show a screen with PIN and the user can't close it until they put the pin code, then export functionality runs and backup key is copied to clipboard, the modal close afterward.

* Checkout this [component](https://github.com/getZION/app/blob/main/src/views/common/Modals/PinCode.tsx) it wraps the PIN in a modal. Maybe you can use it here and prevent closing if that was the default.
* Checkout this [component](https://github.com/getZION/app/blob/main/src/views/account/Security.tsx) as well. It shows how the PIN is wrapped in a modal (different than what i suggested above). 

I hope this helps.

**Comment by ajay-kays** [comment]
New pull request is raised for above mentioned two issues. Please review
